standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/jdbc/TransactionContextManager.java

```diff
@@ -54,11 +57,17 @@ public class TransactionContextManager {
    * The created transaction is wrapped into a {@link TransactionContext} which is {@link AutoCloseable} and allows using
    * the wrapper inside a try-with-resources block.
    *
-   * @param propagation The transaction propagation to use.
+   * @param transactional the transactional definition.
    */
-  public TransactionContext getNewTransaction(int propagation) {
-    TransactionContext context = new TransactionContext(realTransactionManager.getTransaction(
-        new DefaultTransactionDefinition(propagation)), this);
+  public TransactionContext getNewTransaction(Transactional transactional) {
```

Comment:
why not create a parameterless method and use  Propagation.REQUIRED as default?

Reply:
this `transactional` is retrieved from the method annotation, 
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnStore.java#L281
each method might have its own definition, though we only care about the isolation level and propagation now

Reply:
what if we keep the original signature and introduce a change in
````
context = jdbcResource.getTransactionManager().getNewTransaction(
  Optional.ofNullable(transactional).map(Transactional::propagation)
      .orElse(Propagation.REQUIRES_NEW));

....
  public TransactionContext getNewTransaction(Propagation propagation) {
    TransactionContext context = new TransactionContext(realTransactionManager.getTransaction(
        new DefaultTransactionDefinition(propagation.value())), this);
....
}
````

Reply:
so, you decided to get rid of the isolation level? that's fine, but in that case, why do we need 2nd method without the propagation arg?

Reply:
done, removed the 2nd method